### PR TITLE
Improve docs for ComponentElement

### LINKED ISSUE #4505: Confusing comment for ComponentElement
In `framework.dart`, the class level comment for `ComponentElement` says:
```
/// Base class for the instantiation of [StatelessWidget], [StatefulWidget],
/// and [_ProxyWidget] widgets.
```
The exact meaning of this is difficult to grok. `ComponentElement` is not a base class for any of those other classes, so that's not it. Based on the related comment above `Element` that `Elements are the instantiations of Widget configurations`, I think you have a particular meaning in mind for the word instantiation, but it's hard to infer from context.

### PR DESCRIPTION
**Pull request by abarth**
Also, make ProxyWidget and ProxyElement public.

Fixes #4505

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -2162,9 +2169,9 @@ class ErrorWidget extends LeafRenderObjectWidget {
   }
 }
 
-/// Base class for instantiations of widgets that have builders and can be
-/// marked dirty.
+/// An [Element] that can be marked dirty an rebuilt.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: and

File: packages/flutter/lib/src/widgets/framework.dart [hunk 2]
```diff
@@ -2324,9 +2331,12 @@ typedef Widget IndexedWidgetBuilder(BuildContext context, int index);
 // See ComponentElement._builder.
 Widget _buildNothing(BuildContext context) => null;
 
-/// Base class for the instantiation of [StatelessWidget], [StatefulWidget],
-/// and [_ProxyWidget] widgets.
+/// An [Element] that composes other [Element].
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: `other [Element]s.`

File: packages/flutter/lib/src/widgets/framework.dart [hunk 3]
```diff
@@ -2566,13 +2576,15 @@ class StatefulElement extends ComponentElement {
   }
 }
 
-abstract class _ProxyElement extends ComponentElement {
-  _ProxyElement(_ProxyWidget widget) : super(widget) {
+/// Instantiation of [ProxyElement]s.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: for consistency this should be "Base class for instantiations of [ProxyWidget] subclasses."

**Comment by Hixie** [comment]
LGTM

**Pull request merged**